homeassistant/components/foscam/config_flow.py

```diff
@@ -28,14 +33,23 @@
         vol.Required(CONF_USERNAME): str,
         vol.Required(CONF_PASSWORD): str,
         vol.Required(CONF_STREAM, default=STREAMS[0]): vol.In(STREAMS),
+        vol.Required(CONF_RTSP_PORT, default=DEFAULT_RTSP_PORT): int,
     }
 )
 
 
+def camera_unique_id(data):
+    """Generate an unique id for the camera."""
+    # MAC Address is only available with admin accounts so we craft our own unique id
+    return (
+        f"{data[CONF_USERNAME]}_{data[CONF_HOST]}:{data[CONF_PORT]}_{data[CONF_STREAM]}"
```

Comment:
We don't allow ip address as unique id as ip address can change. A unique id must both be unique and stable.

Reply:
In the case we can't really get an unique ID? Because in this case the main issue is that without foscam admin user we cant get any kind of unique ID. Should I just remove it? And if I remove it, how I should handle the entity migration, set it as None?

Reply:
Well I tested removing the unique_id and well as expected now the old configuration.yaml gets imported each time if the user doesnt remove it and also cant change name of the device from the UI so I guess that I would have to add a new field in the configuration for setting it at addition time.

Is there any general advice for this kind of cases where the device doesnt have (or well, can't get) an unique id?

I could get the mac from the discovery protocol (I've been working on it for adding it to Home Assistant https://gist.github.com/skgsergio/e29ef23da7d26d6dbd45d86ec6f3812e) but there are some issues using this like scenarios with restricted traffic or in some network  topologies which makes this unreliable and I can't even ensure this works with all models, which makes this not an option.

Reply:
We can use the `config_entry.entry_id` attribute as unique id for the entities. It'll be available after the config entry has been created.

It's not as good as a serial number or mac address, but it's better than nothing.

Reply:
Thanks @MartinHjelmare, sorted it out, now is using the entry id for the entities, migrates existing ones correctly and can change names and entity ids from the UI.